### DISCUSSION TITLE
[@types/webpack-hot-middleware] using type ICompiler creates conflict

### DISCUSSION THREAD

**Discussion by HenrikSZ** [DefinitelyTyped/DefinitelyTyped#58593] [Issues with a @types package]
This is my first issue report ever, so bear with me please.

When using @types/webpack-hot-middleware with @types/webpack, typescript throws an error:

node_modules/@types/webpack-hot-middleware/index.d.ts(17,23): error TS2724: 'exports' has no exported member named 'ICompiler'. Did you mean 'Compiler'?

This is caused by a conflict in type definitions for the module webpack. More specifically, the package @types/webpack-hot-middleware imports type definitions from webpack. However, it uses an old type definition webpack.ICompiler (seems to be from v4 of @types/webpack). It should instead use webpack.Compiler to not cause a conflict with the types directly from webpack v5.

[a comment by an automated account is not shown]

**Comment by saboya** [comment]
Could be fixed by #58426. I'll review it in the next few days.
